src/index.ts

```diff
@@ -44,11 +65,20 @@ export namespace Validators {
 }
 
 export namespace Utilities {
+
+    export function applyValidators(source: any, target: any, rules: { [key: string]: { type: string } }) {
+        const keys = Object.keys(rules);
+        keys.forEach(key => {
+            if (typeof source[key] !== rules[key].type) throw new TypeError(`${key} must be ${rules[key].type}`);
+            target[key] = source[key];
+        });
```

Comment:
Thanks I hate it

props (ha) for using the proxy for examining the property assignments and applying validation in this manner, very cool and clever, but ugh I hate looking at it

also I'm sorry my brain made me do this

```typescript
const invalidByRuleKey = keys.find(key => typeof source[key] !== rules[key].type);
if (!!invalidByRuleKey ) {
  throw new TypeError(`${invalidByRuleKey } must be ${rules[invalidByRuleKey ].type}`);
}

const checkRuleByProxy = (key: string) => (target[key] = source[key]);
keys.forEach(checkRuleByProxy);

Reply:
Not even sure if this is better code tbh or just more opinionated/I had too much gin before trying to read this stuff